### DISCUSSION TITLE
Is it possible to override GitLab syntax for shorthand links

### DISCUSSION THREAD

**Discussion by gmkado** [facelessuser/pymdown-extensions#2448] [Q&A] [answered]
Referring to https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/#shorthand-links

> The syntax used is actually very similar to GitLab's syntax. GitLab was chosen as its syntax bridges the gaps between the three providers. GitLab and Bitbucket require a pull specific syntax while GitHub does not have one. Also, while GitHub uses #<num> for issues, pulls, and discussions, the idea of using a different symbol for issue types gives us the context we need to generate the correct links for issues vs pulls, etc.

I generate documentation from PR summaries from github, so in those summaries we use the `#` symbol to refer to github links.  I understand this causes issues with how to parse the link (issue, pull, discussion), but if I only care about generating links to pulls can I override the current behavior so `#` generates pull links?

**Comment by facelessuser** [comment, marked as answer]
I'm not sure I understand the request. Can you clarify exactly what you are asking?

GitHub will resolve `#` as an issue. When you click an issue link, GitHub will redirect to PRs or Issues depending on the real context. In short, GitHub does not require `!` for Pull/Merge requests, but can optionally be used if you like the GitLab-like separation.

GitLab is very particular and requires `#` or `!` (within their app) to create links that are either issues or merge requests. An issue URL with a merge request ID will not be redirected merge requests but will go to issues and fail, or find an issue with that number, not matching the intent. So, in order to intelligently tell what a user means in their documents, we follow the same convention.

So, can you explain a bit better why your situation requires a change and how it does not conflict with the above statements?

I generally try to provide a generic approach that fits the broader audience. It can be difficult at times to justify the effort to add in support for niche cases that maybe only 1% of users might utilize. I'm open to considering such work on a case-by-case basis, but I need to understand everything and get a feel for how much it would be used by the wider community.

**Reply by gmkado** [reply]
Ah okay my links were not redirecting and it's because my github repo [did not have issues enabled](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/disabling-issues).  Now it works!
